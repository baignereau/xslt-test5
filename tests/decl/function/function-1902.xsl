<?xml version="1.0"?> 

<?spec xslt#stylesheet-functions?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="2.0"
  xmlns:fos="http://www.w3.org/xpath-functions/spec/namespace"
  xmlns:fn="http://www.w3.org/2005/xpath-functions"
  xmlns:math="http://www.w3.org/2005/xpath-functions/math" exclude-result-prefixes="fos fn math">

  <!-- test that all new functions listed in the F+O specification are implemented (or claim to be available!) in XSLT 2.0 -->
  <!-- This test uses function-catalog.xml as input. This should be updated from time to time
     with the latest version -->

  <xsl:template match="/">
    <xsl:variable name="temp" as="element(f)*">
      <xsl:apply-templates
        select=".//fos:proto[not(../../fos:opermap)][not(../../@prefix=('math', 'map'))]"/>
    </xsl:variable>
    <out>
      <xsl:for-each-group select="$temp, doc('function-1902.out')/out/f" 
         group-by="concat(@name, '|', @arity, '|', @test)">
         <xsl:if test="count(current-group()) ne 2">
           <xsl:copy-of select="current-group()"/>
         </xsl:if>
      </xsl:for-each-group>
    </out>
  </xsl:template>


  <xsl:template match="fos:proto">
    <xsl:variable name="lexname"
      select="if (../../@prefix) then concat(../../@prefix, ':', @name) else @name"/>
    <f name="{$lexname}" arity="{count(fos:arg)}"
      test="{function-available($lexname, count(fos:arg))}"/>
  </xsl:template>
</xsl:stylesheet>
