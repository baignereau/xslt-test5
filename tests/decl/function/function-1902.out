<?xml version="1.0" encoding="UTF-8"?>
<out>
    <f name="node-name" arity="0" test="false"/>
    <f name="node-name" arity="1" test="true"/>
    <f name="fn:nilled" arity="0" test="false"/>
    <f name="fn:nilled" arity="1" test="true"/>
    <f name="fn:string" arity="0" test="true"/>
    <f name="fn:string" arity="1" test="true"/>
    <f name="fn:data" arity="0" test="false"/>
    <f name="fn:data" arity="1" test="true"/>
    <f name="fn:base-uri" arity="0" test="true"/>
    <f name="fn:base-uri" arity="1" test="true"/>
    <f name="fn:document-uri" arity="0" test="false"/>
    <f name="fn:document-uri" arity="1" test="true"/>
    <f name="fn:error" arity="0" test="true"/>
    <f name="fn:error" arity="1" test="true"/>
    <f name="fn:error" arity="2" test="true"/>
    <f name="fn:error" arity="3" test="true"/>
    <f name="fn:trace" arity="2" test="true"/>
    <f name="fn:abs" arity="1" test="true"/>
    <f name="fn:ceiling" arity="1" test="true"/>
    <f name="fn:floor" arity="1" test="true"/>
    <f name="fn:round" arity="1" test="true"/>
    <f name="fn:round" arity="2" test="false"/>
    <f name="fn:round-half-to-even" arity="1" test="true"/>
    <f name="fn:round-half-to-even" arity="2" test="true"/>
    <f name="fn:format-integer" arity="2" test="false"/>
    <f name="fn:format-integer" arity="3" test="false"/>
    <f name="fn:format-number" arity="2" test="true"/>
    <f name="fn:format-number" arity="3" test="true"/>
    <f name="fn:codepoints-to-string" arity="1" test="true"/>
    <f name="fn:string-to-codepoints" arity="1" test="true"/>
    <f name="fn:compare" arity="2" test="true"/>
    <f name="fn:compare" arity="3" test="true"/>
    <f name="fn:codepoint-equal" arity="2" test="true"/>
    <f name="fn:string-join" arity="1" test="false"/>
    <f name="fn:string-join" arity="2" test="true"/>
    <f name="fn:substring" arity="2" test="true"/>
    <f name="fn:substring" arity="3" test="true"/>
    <f name="fn:string-length" arity="0" test="true"/>
    <f name="fn:string-length" arity="1" test="true"/>
    <f name="fn:normalize-space" arity="0" test="true"/>
    <f name="fn:normalize-space" arity="1" test="true"/>
    <f name="fn:normalize-unicode" arity="1" test="true"/>
    <f name="fn:normalize-unicode" arity="2" test="true"/>
    <f name="fn:upper-case" arity="1" test="true"/>
    <f name="fn:lower-case" arity="1" test="true"/>
    <f name="fn:translate" arity="3" test="true"/>
    <f name="fn:encode-for-uri" arity="1" test="true"/>
    <f name="fn:iri-to-uri" arity="1" test="true"/>
    <f name="fn:escape-html-uri" arity="1" test="true"/>
    <f name="fn:contains" arity="2" test="true"/>
    <f name="fn:contains" arity="3" test="true"/>
    <f name="fn:starts-with" arity="2" test="true"/>
    <f name="fn:starts-with" arity="3" test="true"/>
    <f name="fn:ends-with" arity="2" test="true"/>
    <f name="fn:ends-with" arity="3" test="true"/>
    <f name="fn:substring-before" arity="2" test="true"/>
    <f name="fn:substring-before" arity="3" test="true"/>
    <f name="fn:substring-after" arity="2" test="true"/>
    <f name="fn:substring-after" arity="3" test="true"/>
    <f name="fn:matches" arity="2" test="true"/>
    <f name="fn:matches" arity="3" test="true"/>
    <f name="fn:replace" arity="3" test="true"/>
    <f name="fn:replace" arity="4" test="true"/>
    <f name="fn:tokenize" arity="2" test="true"/>
    <f name="fn:tokenize" arity="3" test="true"/>
    <f name="fn:analyze-string" arity="2" test="false"/>
    <f name="fn:analyze-string" arity="3" test="false"/>
    <f name="fn:resolve-uri" arity="1" test="true"/>
    <f name="fn:resolve-uri" arity="2" test="true"/>
    <f name="fn:true" arity="0" test="true"/>
    <f name="fn:false" arity="0" test="true"/>
    <f name="fn:boolean" arity="1" test="true"/>
    <f name="fn:not" arity="1" test="true"/>
    <f name="fn:years-from-duration" arity="1" test="true"/>
    <f name="fn:months-from-duration" arity="1" test="true"/>
    <f name="fn:days-from-duration" arity="1" test="true"/>
    <f name="fn:hours-from-duration" arity="1" test="true"/>
    <f name="fn:minutes-from-duration" arity="1" test="true"/>
    <f name="fn:seconds-from-duration" arity="1" test="true"/>
    <f name="fn:dateTime" arity="2" test="true"/>
    <f name="fn:year-from-dateTime" arity="1" test="true"/>
    <f name="fn:month-from-dateTime" arity="1" test="true"/>
    <f name="fn:day-from-dateTime" arity="1" test="true"/>
    <f name="fn:hours-from-dateTime" arity="1" test="true"/>
    <f name="fn:minutes-from-dateTime" arity="1" test="true"/>
    <f name="fn:seconds-from-dateTime" arity="1" test="true"/>
    <f name="fn:timezone-from-dateTime" arity="1" test="true"/>
    <f name="fn:year-from-date" arity="1" test="true"/>
    <f name="fn:month-from-date" arity="1" test="true"/>
    <f name="fn:day-from-date" arity="1" test="true"/>
    <f name="fn:timezone-from-date" arity="1" test="true"/>
    <f name="fn:hours-from-time" arity="1" test="true"/>
    <f name="fn:minutes-from-time" arity="1" test="true"/>
    <f name="fn:seconds-from-time" arity="1" test="true"/>
    <f name="fn:timezone-from-time" arity="1" test="true"/>
    <f name="fn:adjust-dateTime-to-timezone" arity="1" test="true"/>
    <f name="fn:adjust-dateTime-to-timezone" arity="2" test="true"/>
    <f name="fn:adjust-date-to-timezone" arity="1" test="true"/>
    <f name="fn:adjust-date-to-timezone" arity="2" test="true"/>
    <f name="fn:adjust-time-to-timezone" arity="1" test="true"/>
    <f name="fn:adjust-time-to-timezone" arity="2" test="true"/>
    <f name="format-dateTime" arity="2" test="true"/>
    <f name="format-dateTime" arity="5" test="true"/>
    <f name="format-date" arity="2" test="true"/>
    <f name="format-date" arity="5" test="true"/>
    <f name="format-time" arity="2" test="true"/>
    <f name="format-time" arity="5" test="true"/>
    <f name="resolve-QName" arity="2" test="true"/>
    <f name="QName" arity="2" test="true"/>
    <f name="prefix-from-QName" arity="1" test="true"/>
    <f name="local-name-from-QName" arity="1" test="true"/>
    <f name="namespace-uri-from-QName" arity="1" test="true"/>
    <f name="namespace-uri-for-prefix" arity="2" test="true"/>
    <f name="in-scope-prefixes" arity="1" test="true"/>
    <f name="name" arity="0" test="true"/>
    <f name="name" arity="1" test="true"/>
    <f name="local-name" arity="0" test="true"/>
    <f name="local-name" arity="1" test="true"/>
    <f name="namespace-uri" arity="0" test="true"/>
    <f name="namespace-uri" arity="1" test="true"/>
    <f name="number" arity="0" test="true"/>
    <f name="number" arity="1" test="true"/>
    <f name="lang" arity="1" test="true"/>
    <f name="lang" arity="2" test="true"/>
    <f name="path" arity="0" test="false"/>
    <f name="path" arity="1" test="false"/>
    <f name="root" arity="0" test="true"/>
    <f name="root" arity="1" test="true"/>
    <f name="has-children" arity="0" test="false"/>
    <f name="has-children" arity="1" test="false"/>
    <f name="innermost" arity="1" test="false"/>
    <f name="outermost" arity="1" test="false"/>
    <f name="index-of" arity="2" test="true"/>
    <f name="index-of" arity="3" test="true"/>
    <f name="empty" arity="1" test="true"/>
    <f name="exists" arity="1" test="true"/>
    <f name="distinct-values" arity="1" test="true"/>
    <f name="distinct-values" arity="2" test="true"/>
    <f name="insert-before" arity="3" test="true"/>
    <f name="remove" arity="2" test="true"/>
    <f name="head" arity="1" test="false"/>
    <f name="tail" arity="1" test="false"/>
    <f name="reverse" arity="1" test="true"/>
    <f name="subsequence" arity="2" test="true"/>
    <f name="subsequence" arity="3" test="true"/>
    <f name="unordered" arity="1" test="true"/>
    <f name="zero-or-one" arity="1" test="true"/>
    <f name="one-or-more" arity="1" test="true"/>
    <f name="exactly-one" arity="1" test="true"/>
    <f name="deep-equal" arity="2" test="true"/>
    <f name="deep-equal" arity="3" test="true"/>
    <f name="count" arity="1" test="true"/>
    <f name="avg" arity="1" test="true"/>
    <f name="max" arity="1" test="true"/>
    <f name="max" arity="2" test="true"/>
    <f name="min" arity="1" test="true"/>
    <f name="min" arity="2" test="true"/>
    <f name="sum" arity="1" test="true"/>
    <f name="sum" arity="2" test="true"/>
    <f name="id" arity="1" test="true"/>
    <f name="id" arity="2" test="true"/>
    <f name="element-with-id" arity="1" test="true"/>
    <f name="element-with-id" arity="2" test="true"/>
    <f name="idref" arity="1" test="true"/>
    <f name="idref" arity="2" test="true"/>
    <f name="doc" arity="1" test="true"/>
    <f name="doc-available" arity="1" test="true"/>
    <f name="collection" arity="0" test="true"/>
    <f name="collection" arity="1" test="true"/>
    <f name="uri-collection" arity="0" test="false"/>
    <f name="uri-collection" arity="1" test="false"/>
    <f name="unparsed-text" arity="1" test="true"/>
    <f name="unparsed-text" arity="2" test="true"/>
    <f name="unparsed-text-lines" arity="1" test="false"/>
    <f name="unparsed-text-lines" arity="2" test="false"/>
    <f name="unparsed-text-available" arity="1" test="true"/>
    <f name="unparsed-text-available" arity="2" test="true"/>
    <f name="environment-variable" arity="1" test="false"/>
    <f name="available-environment-variables" arity="0" test="false"/>
    <f name="generate-id" arity="0" test="true"/>
    <f name="generate-id" arity="1" test="true"/>
    <f name="parse-xml" arity="1" test="false"/>
    <f name="parse-xml-fragment" arity="1" test="false"/>
    <f name="serialize" arity="1" test="false"/>
    <f name="serialize" arity="2" test="false"/>
    <f name="position" arity="0" test="true"/>
    <f name="last" arity="0" test="true"/>
    <f name="current-dateTime" arity="0" test="true"/>
    <f name="current-date" arity="0" test="true"/>
    <f name="current-time" arity="0" test="true"/>
    <f name="implicit-timezone" arity="0" test="true"/>
    <f name="default-collation" arity="0" test="true"/>
    <f name="static-base-uri" arity="0" test="true"/>
    <f name="function-lookup" arity="2" test="false"/>
    <f name="function-name" arity="1" test="false"/>
    <f name="function-arity" arity="1" test="false"/>
    <f name="for-each" arity="2" test="false"/>
    <f name="filter" arity="2" test="false"/>
    <f name="fold-left" arity="3" test="false"/>
    <f name="fold-right" arity="3" test="false"/>
    <f name="for-each-pair" arity="3" test="false"/>
    <f name="fn:parse-json" arity="1" test="false"/>
    <f name="fn:parse-json" arity="2" test="false"/>
    <f name="fn:serialize-json" arity="1" test="false"/>
    <f name="fn:serialize-json" arity="2" test="false"/>
</out>
